arangosh/Dump/DumpFeature.cpp

```diff
@@ -513,16 +523,21 @@ int DumpFeature::runDump(std::string& dbName, std::string& errorMsg) {
 
       return TRI_ERROR_CANNOT_WRITE_FILE;
     }
-    meta.close();
+
+    beginEncryption(fd);
 
     std::string const metaString = meta.slice().toJson();
-    if (!TRI_WritePointer(fd, metaString.c_str(), metaString.size())) {
+    bool result = writeData(fd, metaString.c_str(), metaString.size());
+
+    if (!result) {
+      endEncryption(fd);
       TRI_TRACKED_CLOSE_FILE(fd);
       errorMsg = "cannot write to file '" + fileName + "'";
 
       return TRI_ERROR_CANNOT_WRITE_FILE;
     }
 
+    endEncryption(fd);
     TRI_TRACKED_CLOSE_FILE(fd);
   } catch (...) {
```

Comment:
minor improvement:
the statements:
```
  endEncryption(fd);
  TRI_TRACKED_CLOSE_FILE(fd);
```
will be executed inside the `if` condition but also after the if:
```
if (!result) {
  endEncryption(fd);
  TRI_TRACKED_CLOSE_FILE(fd);
  ...
  return ...
}
endEncryption(fd);
TRI_TRACKED_CLOSE_FILE(fd);
```
we should move them before the if then:
```
endEncryption(fd);
TRI_TRACKED_CLOSE_FILE(fd);

if (!result) {
  ...
  return ...
}
```